corehq/apps/app_manager/tests/test_extension_case.py

```diff
@@ -214,6 +224,83 @@ def test_xform_case_block_index_valid_relationship(self):
             )
 
 
+class ExtensionCasesCreateOwnerID(SimpleTestCase):
+    def test_advanced_xform_create_owner_id_with_without_extensions(self):
+        """ Owner id should be automatically set if there are any non-extension indices"""
+
+        # Only extensions
+        advanced_open_action = AdvancedOpenCaseAction.wrap({
+            'case_indices': [
+                {
+                    'tag': 'mother',
+                    'reference_id': 'case',
+                    'relationship': 'extension',
+                },
+                {
+                    'tag': 'father',
+                    'reference_id': 'case',
+                    'relationship': 'extension',
+                }
+            ]
+        })
+        self.assertFalse(autoset_owner_id_for_advanced_action(advanced_open_action))
+
+        # Extension and children
+        advanced_open_action = AdvancedOpenCaseAction.wrap({
+            'case_indices': [
+                {
+                    'tag': 'mother',
+                    'reference_id': 'case',
+                    'relationship': 'extension',
+                },
+                {
+                    'tag': 'father',
+                    'reference_id': 'case',
+                    'relationship': 'child',
+                }
+            ]
+        })
+        self.assertTrue(autoset_owner_id_for_advanced_action(advanced_open_action))
+
+        # No indices
+        advanced_open_action = AdvancedOpenCaseAction()
+        self.assertTrue(autoset_owner_id_for_advanced_action(advanced_open_action))
+
+    def test_advanced_xform_create_owner_id_explicitly_set(self):
+        """When owner_id is explicitly set, don't autoset"""
+        advanced_open_action = AdvancedOpenCaseAction.wrap({
+            'case_properties': {'owner_id': 'owner'},
+            'case_indices': [
+                {
+                    'tag': 'mother',
+                    'reference_id': 'case',
+                    'relationship': 'child'
```

Comment:
Probably not within the scope of this PR, but it would be nice to define `child` and `extension` keywords as constants at some point.